sdk/core/azure-core/src/main/java/com/azure/core/http/policy/UserAgentPolicy.java

```diff
@@ -53,6 +57,35 @@ public UserAgentPolicy(String userAgent) {
         }
     }
 
+    /**
+     * Creates a UserAgentPolicy with the {@code sdkName} and {@code sdkVersion} in the User-Agent header value.
+     *
+     * If the passed configuration contains true for AZURE_TELEMETRY_DISABLED the platform information won't be included
+     * in the user agent.
+     *
+     * @param applicationId User specified application Id.
+     * @param sdkName Name of the client library.
+     * @param sdkVersion Version of the client library.
+     * @param version {@link ServiceVersion} of the service to be used when making requests.
+     * @param configuration Configuration store that will be checked for the AZURE_TELEMETRY_DISABLED.
+     */
+    public UserAgentPolicy(String applicationId, String sdkName, String sdkVersion, Configuration configuration,
+                           ServiceVersion version) {
+        boolean telemetryDisabled = configuration.get(Configuration.PROPERTY_AZURE_TELEMETRY_DISABLED, false);
```

Comment:
Should `null` check the passed configuration.

Reply:
done